bokehjs/src/coffee/core/has_props.coffee

```diff
@@ -405,6 +405,10 @@ export class HasProps extends Backbone.Model
       # this skips optional properties like radius for circles
       if (prop.optional || false) and prop.spec.value == null and (name not of @_set_after_defaults)
         continue
+      if prop.spec.transform
+        if not prop.spec.transform._listeners[this._listenId]
```

Comment:
It might also be worth investigating if this could be done here in this loop instead:

https://github.com/bokeh/bokeh/blob/15037b3e4e9fc7aa4eb061a94acaec48af897af1/bokehjs/src/coffee/core/has_props.coffee#L132-L133

That's already looping over all the properties so it seems like a nice place to set things up. Also it only happens once so I dont think the check for listeners would be needed